x-pack/plugins/observability_solution/synthetics/server/common/pings/monitor_status_heatmap.ts

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { SyntheticsEsClient } from '../../lib';
+
+export async function queryMonitorHeatmap({
+  syntheticsEsClient,
+  from,
+  to,
+  monitorId,
+  location,
+  intervalInMinutes,
+}: {
+  syntheticsEsClient: SyntheticsEsClient;
+  from: number | string;
+  to: number | string;
+  monitorId: string;
+  location: string;
+  intervalInMinutes: number;
+}) {
+  return syntheticsEsClient.search({
+    body: {
+      size: 0,
+      query: {
+        bool: {
+          filter: [
+            {
+              exists: {
+                field: 'summary',
+              },
+            },
+            {
+              range: {
+                '@timestamp': {
+                  gte: from,
+                  lte: to,
+                },
+              },
+            },
+            {
+              term: {
+                'monitor.id': monitorId,
+              },
+            },
+          ],
+        },
+      },
+      sort: [
+        {
+          '@timestamp': {
+            order: 'desc',
+          },
+        },
+      ],
+      post_filter: {
+        terms: {
+          'observer.geo.name': [location],
+        },
+      },
```

Comment:
why is this a post_filter?

Reply:
eh, because I adapted this from the `query_pings` API and didn't notice I was applying a `post_filter` when the goal is to use aggs. Well-spotted, we should change this to a filter so we make sure the aggregations accurately reflect the data we want.